pytorch_lightning/trainer/trainer.py

```diff
@@ -143,7 +143,8 @@ def __init__(
         distributed_backend: Optional[str] = None,
         move_metrics_to_cpu: bool = False,
         multiple_trainloader_mode: str = 'max_size_cycle',
-        stochastic_weight_avg: bool = False
+        stochastic_weight_avg: bool = False,
+        serialize_checkpoint_loading: bool = False
```

Comment:
Do you think a trainer flag is necessary here? Is it too slow to always serialize?

What about us making the choice by comparing the ram available and model size?

Reply:
It can potentially be unsafe to always serialize, specifically if you are doing any NCCL communications in `on_load_checkpoint` and assume they are happening at the same time on all hosts - with serialization those will deadlock, hence having this off by default.